vaadin-side-nav-flow-parent/vaadin-side-nav-flow/src/main/java/com/vaadin/flow/component/sidenav/SideNav.java

```diff
@@ -151,6 +162,65 @@ protected void onAttach(AttachEvent attachEvent) {
         checkFeatureFlag();
     }
 
+    /**
+     * Gets the internationalization object previously set for this component.
+     * <p>
+     * Note: updating the object content that is gotten from this method will
+     * not update the lang on the component if not set back using
+     * {@link SideNav#setI18n(SideNavI18n)}
+     *
+     * @return the i18n object. It will be <code>null</code>, If the i18n
+     *         properties weren't set.
+     */
+    public SideNavI18n getI18n() {
+        return i18n;
+    }
+
+    /**
+     * Updates the i18n settings in the web component. Merges the
+     * {@link SideNavI18n} settings with the current / default settings of the
+     * web component.
+     *
+     * @param i18n
+     *            the internationalized properties, not <code>null</code>
+     */
+    public void setI18n(SideNavI18n i18n) {
+        Objects.requireNonNull(i18n,
+                "The i18N properties object should not be null");
+        this.i18n = i18n;
+        requestI18nUpdate();
+    }
+
+    private void requestI18nUpdate() {
+        getUI().ifPresent(ui -> {
```

Comment:
This logic seems to be copied from the `DatePicker` which has extra complexity.
Consider using a simpler approach like in `Avatar` and `AvatarGroup`:

```java
getElement().setPropertyJson("i18n", JsonSerializer.toJson(i18n));
```

Reply:
Yes, I checked a few implementations and combined some of their approaches. I simplified the approach with the latest commit. However, there are lots of reused logic regarding i18n, we can maybe consider introducing an interface like `hasInternationalization` in the future.

Reply:
Makes sense, thank you. Let's consider this when we get to improving `i18n` (which would be a separate epic).